java/src/jmri/jmrix/loconet/SlotManager.java

```diff
@@ -781,6 +782,55 @@ protected void respondToAddrRequest(LocoNetMessage m, int i) {
         }
     }
 
+    /**
+     * If it is a slot being sent COMMON,
+     *  after a delay, get the new status of the slot
+     * If it is a true slot move, not dispatch or null
+     *  after a delay, get the new status of the from slot, which varies by CS.
+     *  the to slot should come in the reply.
+     * @param m a LocoNet message
+     * @param i the slot to which it is directed
+     */
+    protected void getMoreDetailsForSlot(LocoNetMessage m, int i) {
+        // is called any time a LocoNet message is received.
+        // sets up delayed slot read to update our effected slots to match the CS
+        if (m.getOpCode() == LnConstants.OPC_SLOT_STAT1 &&
+                ((m.getElement(2) & LnConstants.LOCOSTAT_MASK) == LnConstants.LOCO_COMMON ) ) {
+            // Changing a slot to common. Depending on a CS and its OpSw, and throttle speed 
+            // it could have its status changed a number of ways.
+            sendReadSlotDelayed(i,100);
+        } else if (m.getOpCode() == LnConstants.OPC_MOVE_SLOTS) {
+            // if a true move get the new from slot status
+            // the to slot status is sent in the reply
+            int slotTwo;
+            slotTwo = m.getElement(2);
+            if (i != 0 && slotTwo != 0) {
+                sendReadSlotDelayed(i,100);
+            }
+       }
+    }
+
+    /**
+     * Scedule a delayed slot read.
+     * @param slotNo - the slot.
+     * @param delay - delay in msecs.
+     */
+    protected void sendReadSlotDelayed(int slotNo, long delay) {
+        java.util.TimerTask meterTask = null;
+        meterTask = new java.util.TimerTask() {
```

Comment:
I expected Spotbugs to fail on this. `meterTask` is first assigned to `null` and then assigned to a `TimerTask`. Spotbugs normally doesn't accept this since the null value is not used.

Reply:
@Pugwash1 
I think it would be good to fix this before merge of this PR. I think Spotbugs should have failed on this and even if it didn't now, it may fail later on this.

`java.util.TimerTask meterTask = new java.util.TimerTask() {`